hudi-client/src/main/java/org/apache/hudi/table/action/commit/BaseCommitActionExecutor.java

```diff
@@ -231,15 +231,7 @@ protected void finalizeWrite(String instantTime, List<HoodieWriteStat> stats, Ho
   }
 
   private void updateMetadataAndRollingStats(HoodieCommitMetadata metadata, List<HoodieWriteStat> writeStats) {
-    // 1. Look up the previous compaction/commit and get the HoodieCommitMetadata from there.
-    // 2. Now, first read the existing rolling stats and merge with the result of current metadata.
-
-    // Need to do this on every commit (delta or commit) to support COW and MOR.
-    for (HoodieWriteStat stat : writeStats) {
-      String partitionPath = stat.getPartitionPath();
-      // TODO: why is stat.getPartitionPath() null at times here.
-      metadata.addWriteStat(partitionPath, stat);
-    }
+    writeStats.forEach(stat -> metadata.addWriteStat(stat.getPartitionPath(), stat));
```

Comment:
Same case for this also. Can you remove this method and move "writeStats.forEach(stat -> metadata.addWriteStat(stat.getPartitionPath(), stat));" to caller

Reply:
Good suggestion, fixed